### PR TITLE
Feature/add redis

### PR DESCRIPTION
**Pull request by LoneXDii**

### CODE REVIEW

**Comment by LoneXDii** [comment]
I used a force push because of the git rebase (I needed to do this due to merges with previous PRs edits into dev)

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+        if(cartId is null)
+        {
+            cartId = Guid.NewGuid().ToString();
+            _httpContext.Response.Cookies.Append("CartId", cartId, new CookieOptions
+            {
+                Expires = DateTimeOffset.UtcNow.AddDays(30), 
+                HttpOnly = true, 
+                Secure = true, 
+            });
+        }
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: maybe let's move this to a separate method? it's used also in SaveCartAsync

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+
+        var userId = _httpContext.User.FindFirst("Id")?.Value;
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs:22]: you can do it in the beginning of the method and use userId in finding of cartId

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        }
+
+        var cartJson = JsonSerializer.Serialize(cart);
+
+        var options = new DistributedCacheEntryOptions
+        {
+            AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(30)
+        };
+
+        if (_httpContext.User.FindFirst("Id")?.Value is not null)
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: the same, you can get userId in the beginning and then use it

File: UserService/UserService.API/Controllers/UsersController.cs [hunk 4]
```diff
@@ -80,12 +82,30 @@ public async Task<ActionResult<UserInfoDTO>> GetUserInfo()
     }
 
     [HttpGet]
-    [Route("info/id={userId}")]
+    [Route("info/id")]
```
Comment (UzheNeKate) [inline UserService/UserService.API/Controllers/UsersController.cs]: it's better to make it {userId}/info

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+
+        if (userId is not null)
+        {
+            options.AbsoluteExpirationRelativeToNow = TimeSpan.FromDays(1);
+        }
+
+        await _cache.SetStringAsync(cartId, cartJson, options, cancellationToken);
+    }
+
+    private string GetCartId()
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: you can pass userId as a parameter

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 6]
```diff
@@ -0,0 +1,95 @@
...
+
+        var options = new DistributedCacheEntryOptions
+        {
+            AbsoluteExpirationRelativeToNow = TimeSpan.FromMinutes(30)
+        };
+
+        if (userId is not null)
+        {
+            options.AbsoluteExpirationRelativeToNow = TimeSpan.FromDays(1);
+        }
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: maybe let's combine this? 
AbsoluteExpirationRelativeToNow = userId is null ? TimeSpan.FromMinutes(30) : TimeSpan.FromDays(1)

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 7]
```diff
@@ -0,0 +1,95 @@
...
+
+                    var options = new DistributedCacheEntryOptions
+                    {
+                        AbsoluteExpirationRelativeToNow = TimeSpan.FromDays(1)
+                    };
+
+                    await _cache.SetStringAsync(userId, oldCart, options, cancellationToken);
+                }
+            }
+        }
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: can you explain this logic, please?

File: UserService/UserService.API/Controllers/AccountController.cs [hunk 8]
```diff
@@ -11,7 +11,7 @@
 
 namespace UserService.API.Controllers;
 
-[Route("api/[controller]")]
+[Route("api/account/")]
```
Comment (UzheNeKate) [inline UserService/UserService.API/Controllers/AccountController.cs]: please use plural nouns

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs (hunk 7 above)
Comment (LoneXDii) [reply OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: This code is used to assign a cart to a userId if the user is authorized. I'm checking if there is a cartId in the cookies, and if it exists, I'm deleting it from the cookies and assigning this cart to the userId

Comment (UzheNeKate) [reply OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: what do you think about moving it to a separate method?

File: UserService/UserService.DAL/Services/TemporaryStorage/CacheService.cs [hunk 9]
```diff
@@ -0,0 +1,53 @@
...
+        return code;
+    }
+
+    public async Task<string?> GetEmailByResetPasswordCodeAsync(string code)
+    {
+        var email = await _cache.GetStringAsync($"password{code}");
+
+        return email;
+    }
+    public async Task<string> SetResetPasswordCodeAsync(string email)
```
Comment (UzheNeKate) [inline UserService/UserService.DAL/Services/TemporaryStorage/CacheService.cs]: please split methods with empty line

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 10]
```diff
@@ -0,0 +1,93 @@
...
+
+        return cartId;
+    }
+
+    private async Task AssignCartToUserIdAsync(string userId, CancellationToken cancellationToken)
+    {
+        var oldCartId = _httpContext.Request.Cookies["CartId"];
+
+        if (oldCartId is not null)
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: you can revert this if

File: OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs [hunk 11]
```diff
@@ -0,0 +1,93 @@
...
+        {
+            var oldCart = await _cache.GetStringAsync(oldCartId, cancellationToken);
+
+            _httpContext.Response.Cookies.Delete("CartId");
+
+            if (oldCart is not null)
```
Comment (UzheNeKate) [inline OrderService/OrderService.Infrastructure/Services/RedisStorageService.cs]: and this

**Review by UzheNeKate — APPROVED** [review]

**Pull request merged**